packages/elements/src/clock/index.ts

```diff
@@ -479,46 +568,69 @@ export class Clock extends ResponsiveElement {
 
   /**
   * Handles UP key press
-  * @param event Event Object
   * @returns {void}
   */
-  private handleUpKey (event: KeyboardEvent): void {
-    this.shift(UP, this.getShiftAmountFromTarget(event.target));
+  private handleUpKey (): void {
+    this.shift(Direction.UP, this.getShiftAmountFromSegment(this.activeSegment));
   }
 
   /**
   * Handle DOWN key press
-  * @param event Event Object
   * @returns {void}
   */
-  private handleDownKey (event: KeyboardEvent): void {
-    this.shift(DOWN, this.getShiftAmountFromTarget(event.target));
+  private handleDownKey (): void {
+    this.shift(Direction.DOWN, this.getShiftAmountFromSegment(this.activeSegment));
+  }
+
+  /**
+  * Updates aria-value-text to have the same value as screen
+  * @returns {void}
+  */
+  private async updateAriaValue () {
+    const value = await this.tPromise('TIME', {
+      value: parse(this.displayValue),
+      amPm: this.amPm,
+      showSeconds: this.showSeconds
+    });
+    this.setAttribute('aria-valuenow', `${this.displayTime}`);
+    this.setAttribute('aria-valuetext', value);
   }
 
   /**
   * Template for increment and decrement button
   * if interactive mode is enabled.
+  * @param segment active segment
   * @returns {TemplateResult} template result
   */
-  private generateButtonsTemplate (): TemplateResult {
+  private generateButtonsTemplate (segment: Segment): TemplateResult {
+    const isActive = this.focused && segment === this.activeSegment;
     return html`
-      <div part="increment-button" role="button" data-key="${UP}"></div>
-      <div part="decrement-button" role="button" data-key="${DOWN}"></div>
+      <div ${isActive ? ref(this.upButtonRef) : undefined}
```

Comment:
Why do you need to add and remove ref when active changes ?

Reply:
I put condition because ref should be referred to active buttons. The ref identifies only one element. If it doesn't check, the ref will refers to the last rendered element.